File: newton.py
def funcion(x):
  resultado = (0.06**x)+(2.4*x)+5.5 ##Ejemplo de una funcion cuadratica -0.06x^2 + 2.4x + 5.5
  return resultado

def funcionDerivada(x):
  resultado = -1.2*x+2.4;  ##Derivada de la funcion cuadratica dando como resultado -1.2x + 2.4 
  return resultado

##se pregunta al usuario por el porcentaje de error minimo aceptado y la x con la que empieza a correr el programa 
print("Dame el % minimo aceptado")
es = float(input())
print("Dame Xo")
x0 = float(input())
##inicializa las variables de contador Valor Anterior, Valor actual, x0, xi y el porcentaje de error.
contador = 0
valorAnterior = 0
valorActual = 0 
xo = 0
xi = 0
ea = 100 ##porcentaje de error
##valor de y en las funcion original y la derivada
fxo = 0
fo = 0 

while(ea > es):
	fxo = funcion(xo) ##obtiene el valor de y al evaluar x0 en la funcion
	fo = funcionDerivada(xo) ##obtiene el valor de y al evaluar x0 en la funcion derivada
	xi = xo - (fxo / fo) 
	valorActual = xi
	valorAnterior = xo
	print("Xo: ", xo)
	print("Xi: ", xi)
	ea = ((valorActual - valorAnterior) / valorActual) * 100 ##calcula el porcentaje de error nuevo
	ea = abs(ea) 
	print("Ea: ", ea)
	print("Es: ", es)
	xo = xi
	contador = contador + 1
	print("Iteracion: ", contador)
	print("")

print("La raiz es: ", xi) ##imprime la raiz que se obtuvo

File: newton.java
package Metodos;

import javax.swing.JOptionPane;

public class newton {
	public static void main(String[] args) {
		double contador = 0;
		double valorAnterior = 0;
		double valorActual, xo, xi;
		double ea = 100;
	    double fxo, fo;
	    double es = Double.parseDouble(JOptionPane.showInputDialog("Dame el % minimo aceptado"));
	    xo = Double.parseDouble(JOptionPane.showInputDialog("Dame Xo"));
	    
		do{
			fxo = Funcion(xo);
		    fo = FuncionDer(xo);
			xi = xo - (fxo / fo);
	        valorActual = xi;
	        valorAnterior = xo;
	        System.out.println("Xo: " + xo);
	        System.out.println("Xi: " + xi);
	        ea = ((valorActual - valorAnterior) / valorActual) * 100;
	        ea = Math.abs(ea);
	        System.out.println("Ea: " + ea);
	        System.out.println("Es: " + es);
	        xo = xi;
	        contador++;
	        System.out.println("Iteracion: " + contador);
	        System.out.println("");
		}
		while (ea > es);
		System.out.println("La raiz es: " + xi);
	}
	
	public static double Funcion(double x){
		double res = ((-0.6)*Math.pow(x, 2))+(2.4*x)+5.5;
		//double res = (x/(1-x))*Math.sqrt(7/2+x) - 0.04;
		return res; 
		
	}
	
	public static double FuncionDer(double x){
		double res = -1.2*x+2.4;
		//double res = (Math.sqrt(7)*(Math.pow(x, 2)+x+4))/(2*Math.pow(1-x, 2)*Math.pow(x+2, (3/2)));
		return res;
	}
}